balancer/grpclb/grpclb.go

```diff
@@ -245,6 +259,34 @@ type lbBalancer struct {
 //  - does two layer roundrobin pick otherwise.
 // Caller must hold lb.mu.
 func (lb *lbBalancer) regeneratePicker() {
+	// Keep pickfirst logic in a separate if statement, even though it means
+	// some duplicate code.
```

Comment:
I think you could avoid some duplicate code by moving the handling of transient failure / connecting above this `if`.

Reply:
Done